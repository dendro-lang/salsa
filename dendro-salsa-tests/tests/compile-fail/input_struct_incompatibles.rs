#[salsa::jar(db = Db)]
struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);

trait Db: salsa::DbWithJar<Jar> {}

#[salsa::input(jar = Jar, return_ref)]
struct InputWithRetRef (u32);

#[salsa::input(jar = Jar, specify)]
struct InputWithSpecify (u32);

#[salsa::input(jar = Jar, no_eq)]
struct InputNoWithEq (u32);

#[salsa::input(jar = Jar, db = Db)]
struct InputWithDb (u32);

#[salsa::input(jar = Jar, recover_fn = recover)]
struct InputWithRecover (u32);

#[salsa::input(jar = Jar, lru =12)]
struct InputWithLru (u32);

fn main() {}