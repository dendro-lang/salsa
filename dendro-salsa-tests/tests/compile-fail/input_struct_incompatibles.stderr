error: `return_ref` option not allowed here
 --> tests/compile-fail/input_struct_incompatibles.rs:6:27
  |
6 | #[salsa::input(jar = Jar, return_ref)]
  |                           ^^^^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/input_struct_incompatibles.rs:9:27
  |
9 | #[salsa::input(jar = Jar, specify)]
  |                           ^^^^^^^

error: `no_eq` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:12:27
   |
12 | #[salsa::input(jar = Jar, no_eq)]
   |                           ^^^^^

error: `db` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:15:27
   |
15 | #[salsa::input(jar = Jar, db = Db)]
   |                           ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/input_struct_incompatibles.rs:18:27
   |
18 | #[salsa::input(jar = Jar, recover_fn = recover)]
   |                           ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:21:27
   |
21 | #[salsa::input(jar = Jar, lru =12)]
   |                           ^^^

error[E0412]: cannot find type `InputWithRetRef` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:12
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |            ^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputWithSpecify` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:29
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |                             ^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputNoWithEq` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:47
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |                                               ^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputWithDb` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:62
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |                                                              ^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputWithRecover` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:75
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |                                                                           ^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InputWithLru` in this scope
 --> tests/compile-fail/input_struct_incompatibles.rs:2:93
  |
2 | struct Jar(InputWithRetRef, InputWithSpecify, InputNoWithEq, InputWithDb, InputWithRecover, InputWithLru);
  |                                                                                             ^^^^^^^^^^^^ not found in this scope
